modules/client/src/client-common/context.ts

```diff
@@ -211,48 +232,61 @@ export class Context {
    *
    * @var ipfs
    *
-   * @returns {IpfsClient[] | undefined}
+   * @returns {IpfsClient[]}
    *
    * @public
    */
-  get ipfs(): IpfsClient[] | undefined {
-    return this.state.ipfs || defaultState.ipfs;
+  get ipfs(): IpfsClient[] {
+    return this.state.ipfs;
   }
 
   /**
    * Getter for the GraphQL client
    *
    * @var graphql
    *
-   * @returns {GraphQLClient[] | undefined}
+   * @returns {GraphQLClient[]}
    *
    * @public
    */
-  get graphql(): GraphQLClient[] | undefined {
-    return this.state.graphql || defaultState.graphql;
+  get graphql(): GraphQLClient[] {
+    return this.state.graphql;
   }
 
-  // DEFAULT CONTEXT STATE
-  static setDefault(params: Partial<ContextParams>) {
-    if (params.daoFactoryAddress) {
-      defaultState.daoFactoryAddress = params.daoFactoryAddress;
+  // INTERNAL HELPERS
+  private static resolveNetwork(
+    networkish: Networkish,
+    ensRegistryAddress?: string,
+  ): Network {
+    const network = getNetwork(networkish);
+    const networkName = network.name as SupportedNetworks;
+    if (!SupportedNetworksArray.includes(networkName)) {
+      throw new UnsupportedNetworkError(networkName);
     }
-    if (params.signer) {
-      defaultState.signer = params.signer;
+    if (ensRegistryAddress) {
+      if (!isAddress(ensRegistryAddress)) {
+        console.warn("invalid address, using default");
```

Comment:
Why not throwing an error? If it was empty, it's fine to fall back to the default. But if it's incorrect, the developer should be aware of that

Reply:
okay, I lef it as an option but we can throw an error